### PR TITLE
settings: Add color-scheme key

### PR DESCRIPTION
**Pull request by Exalm**
Specify a key for getting the system's preferred color scheme.

Fixes https://github.com/flatpak/xdg-desktop-portal/issues/629

### CODE REVIEW

File: data/org.freedesktop.impl.portal.Settings.xml [hunk 1]
```diff
@@ -27,9 +27,27 @@
...
+        <term>org.freedesktop.appearance color-scheme u</term>
+        <listitem><para>
+          Indicates the system's preferred color scheme.
+          Supported values are:
+          <simplelist>
+            <member>0: No preference</member>
+            <member>1: Prefer dark appearance</member>
+            <member>2: Prefer light appearance</member>
+          </simplelist>
+          Unknown values should be considered same as 0.
```
Comment (GeorgesStavracas) [inline data/org.freedesktop.impl.portal.Settings.xml:43]: `Unknown values should be treated as 0 (no preference).`

File: data/org.freedesktop.portal.Settings.xml [hunk 2]
```diff
@@ -27,9 +27,27 @@
...
+        <term>org.freedesktop.appearance color-scheme u</term>
+        <listitem><para>
+          Indicates the system's preferred color scheme.
+          Supported values are:
+          <simplelist>
+            <member>0: No preference</member>
+            <member>1: Prefer dark appearance</member>
+            <member>2: Prefer light appearance</member>
+          </simplelist>
+          Unknown values should be considered same as 0.
```
Comment (GeorgesStavracas) [inline data/org.freedesktop.portal.Settings.xml:43]: `Unknown values should be treated as 0 (no preference).`

**Comment by TingPing** [comment]
Looks fine to merge once implementations exist.

**Comment by Exalm** [comment]
https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/merge_requests/12

Do we need an equivalent `xdg-desktop-portal-gtk` change? On one hand it's fairly GNOME-specific, but `-gtk` already has plenty of GNOME things in it, since `xdg-desktop-portal-gnome` exists now, not sure what's the policy regarding that.

**Comment by Exalm** [comment]
elementary implementation: https://github.com/elementary/settings-daemon/pull/38

**Comment by TingPing** [comment]
I'll leave the -gtk question to @GeorgesStavracas

**Comment by GeorgesStavracas** [comment]
> Do we need an equivalent xdg-desktop-portal-gtk change?

I would say, yes.

**Comment by Exalm** [comment]
> I would say, yes.

Ok - https://github.com/flatpak/xdg-desktop-portal-gtk/pull/354

**Comment by lilianmoraru** [comment]
@Exalm, @GeorgesStavracas, @TingPing
Could you please also add support for High Contrast?
Microsoft does it very nicely: https://support.microsoft.com/en-us/windows/use-high-contrast-mode-in-windows-10-fedc744c-90ac-69df-aed5-c8a90125e696 , allowing users that maybe have problems seeing a particular color to switch to a different one.
Here they describe the event that apps will receive and how to read the user's desired colors, in order to avoid hardcoding them: https://docs.microsoft.com/en-us/windows/win32/w8cookbook/high-contrast-mode#solution

**Comment by Exalm** [comment]
We have it in GNOME, but I consider it a separate issue when it comes to having a standardized one - for example, elementary doesn't have it.

**Comment by lilianmoraru** [comment]
GNOME has a separate accessibility toggle with no dedicated D-Bus connection to listen on, for apps to handle it(and missing documentation).
It needs a more official/portable way of doing it(IMO). I thought that once you are here, this could be handled as well.

**Comment by Exalm** [comment]
I mean until recently (still is in 41) it was a theme - not even a flag. One thing at a time :p

**Comment by WhyNotHugo** [comment]
I'm trying to read this setting with non-flatpak'd apps (in my case, d-feet), but I get:

`` `
('g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.AccessDenied: '
 'Portal operation not allowed: Unable to open /proc/147042/root (9)')
```

Are native applications supposed to use a different mechanism to retrieve this value from the portal?

**Comment by swick** [comment]
IMO no, if it's not working for some reason that's a bug.

**Comment by rusty-snake** [comment]
Screenshot_2022-01-13_15-12-49-fs8

**Comment by WhyNotHugo** [comment]
@rusty-snake Reported in #737

**Pull request merged**